### PR TITLE
PF-2407: Update tps version and also add connected test

### PR DESCRIPTION
**Pull request by yuhuyoyo**

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java [hunk 1]
```diff
@@ -443,6 +446,79 @@ public void listValidRegions() throws Exception {
...
+
+    assertTrue(result.isUpdateApplied());
+    ApiWorkspaceDescription updatedWorkspace =
+        mockMvcUtils.getWorkspace(userAccessUtils.defaultUserAuthRequest(), workspace.getId());
+    assertEquals(1, updatedWorkspace.getPolicies().size());
+
+    mockMvcUtils.updatePoliciesExpect(
+        userAccessUtils.defaultUserAuthRequest(),
+        workspace.getId(),
+        HttpStatus.SC_INTERNAL_SERVER_ERROR,
```
Comment (ddietterich) [inline service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java:500]: I'm surprised this is a 500 error. Where is that coming from?
If this is a conflict, we should be returning success with conflict information.

So see, the test already has found a bug! :)

Not sure if we should apply the test until we get that straighted out. wdyt?

Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java:500]: oh ok. I wasn't sure what's the right behavior is.

the fix should be in TPS right? we are getting 409 from TPS and then wrap it in a 500 in TpsApidispatch. are you saying updatePao should return 200 for ENFORCE_CONFLICT and put conflict information in the response?

Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java]: 409 are wrapped now.

File: service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java [hunk 2]
```diff
@@ -443,6 +446,77 @@ public void listValidRegions() throws Exception {
...
 
+  @Test
+  @EnabledIf(expression = "${feature.tps-enabled}", loadContext = true)
+  public void updatePolicies_tpsEnabledAndPolicyUpdated() throws Exception {
+    ApiWorkspaceDescription workspaceWithoutPolicy =
+        mockMvcUtils.getWorkspace(userAccessUtils.defaultUserAuthRequest(), workspace.getId());
+    assertTrue(workspaceWithoutPolicy.getPolicies().isEmpty());
+
+    // add attributes
+    String usRegion = "US";
```
Comment (mikenev) [inline service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java:457]: This looks ok, but I'm not sure why we have capital "US" here. That's not one of the regular regions. And we use lower case "us" on 487. Maybe we should use the "usa" region?

Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java:457]: it does allow me to update policy attributes to US though. i thought it's tps being smart. is it a bug?

**Review by mikenev — APPROVED** [review]

File: service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java (hunk 2 above)
Comment (mikenev) [reply service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java]: We don't currently validate the value of the constraint. So you can set it to US if you want. If you attempt to merge that policy with something later or retrieve valid locations for it, then you won't be able to merge any policies or find any valid locations.
Maybe we should be validating that input.

**Review by ddietterich — APPROVED** [review]

File: service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java (hunk 2 above)
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java]: agree with validating the input.

**Pull request merged**